bin/yang_get_stats.py

```diff
@@ -108,185 +137,164 @@ def start_process(self):
+                self._write_dictionary_file_in_json(
+                    prefix_info['compilation_stats'],
+                    self.stats_path,
+                    prefix_info['stats_filename'],
                 )
-                with open(json_history_file, 'w') as f:
-                    json.dump(yang_page_compilation_stats, f)
-                self._write_dictionary_file_in_json(yang_page_compilation_stats, self.stats_path, filename)
-
-    def gather_ietf_yang_out_of_rfc_compilation_stats(self):
-        print(f'\nLooking at the files starting with: {self.IETF_YANG_OUT_OF_RFC_PREFIX}')
-        print('FILENAME: NUMBER OF DAYS SINCE EPOCH, NUMBER OF YANG MODELS IN RFCS')
-        json_history_file = os.path.join(self.backup_directory, f'{self.IETF_YANG_OUT_OF_RFC_PREFIX}history.json')
-        yang_page_compilation_stats = self._load_compilation_stats_from_history_file(json_history_file)
-        for filename in self._file_names_containing_keyword(self.IETF_YANG_OUT_OF_RFC_PREFIX):
-            path_to_file = os.path.join(self.backup_directory, filename)
-            rfc_result = 0
-            with open(path_to_file, 'r') as f:
-                for line in f:
-                    if '.yang' in line:
-                        rfc_result += 1
-            extracted_date = self._extract_date_from_filename(filename)
-            if (datetime.date.today() - extracted_date).days > 30:
-                self.remove_old_html_file_paths.append(path_to_file)
-            yang_page_compilation_stats[date2num(extracted_date)] = {'total': rfc_result}
-        if self.days == -1:
-            with open(json_history_file, 'w') as f:
-                json.dump(yang_page_compilation_stats, f)
-            self._write_dictionary_file_in_json(
-                yang_page_compilation_stats,
-                self.stats_path,
-                f'{self.IETF_YANG_OUT_OF_RFC_PREFIX[:-1]}Stats.json',
-            )
 
-    def _load_compilation_stats_from_history_file(self, json_history_file: str) -> dict:
-        compilation_stats = {}
-        if os.path.isfile(json_history_file):
-            with open(json_history_file, 'r') as f:
-                compilation_stats_temp = json.load(f)
-            for key, value in compilation_stats_temp.items():
-                compilation_stats[float(key)] = value
-            del compilation_stats_temp
-        return compilation_stats
+    def _filename_contains_prefix(self, filename: str) -> t.Optional[str]:
+        filename = filename.lower()
+        # We have to find the longest one in order to prevent incorrect results, here's the explanation:
+        # filename = IETFYANGPageMain_2022_01_01.html -> prefixes YANGPageMain_ and IETFYANGPageMain_ are both present
+        # in this filename but only IETFYANGPageMain_ is the correct one
+        longest_prefix = None
+        for prefix in self.ALL_PREFIXES:
+            if prefix.lower() not in filename:
```

Comment:
`filename.startswith(prefix.lower())` would probably be better here, assuming we don't rely on matching "prefixes" in the middle of the filename.